galsim/base.py

```diff
@@ -2181,6 +2181,105 @@ def getScaleRadius(self):
+    thrown if you provide a value outside that range.  These values are chosen to match the values
+    investigated by the above paper.  At and below nu=0.0, the surface brightness profile diverges
+    at the origin, which currently affects our ability to perform photon-shooting or `real_space`
+    convolutions.
+
+    The Spergel profile is the Fourier transform of the Moffat profile, and hence has an analytic
+    Fourier transform if not truncated.  If truncation is used, however, then the calculations take
+    advantage of Hankel transform tables that are precomputed for a given value of nu and truncation
+    radius when the Spergel profile is initialized.  Making additional objects with the same nu and
+    truncation radius can therefore be many times faster than making objects with different values
+    of nu and truncation radius that have not been used before.  Moreover, these Hankel transforms
+    are only cached for a maximum of 100 different nu/truncation radius pairs at a time.  For this
+    reason, for large sets of simulations, it is worth considering the use of only discrete
+    nu/truncation radius pairs rather than allowing these to vary continuously.
+
+    A Spergel can be initialized using one (and only one) of two possible size parameters:
+    `scale_radius` or `half_light_radius`.  Exactly one of these two is required.
+
+    @param nu               The Spergel index, nu.
+    @param half_light_radius  The half-light radius of the profile.  Typically given in arcsec.
+                            [One of `scale_radius` or `half_light_radius` is required.]
+    @param scale_radius     The scale radius of the profile.  Typically given in arcsec.
+                            [One of `scale_radius` or `half_light_radius` is required.]
+    @param flux             The flux (in photons) of the profile. [default: 1]
+    @param trunc            An optional truncation radius at which the profile is made to drop to
+                            zero.  [default: 0, indicating no truncation]
+    @param flux_untruncated Should the provided `flux` and `half_light_radius` refer to the
+                            untruncated profile? See below for more details. [default: False]
+    @param gsparams         An optional GSParams argument.  See the docstring for GSParams for
+                            details. [default: None]
+
+    Flux of a truncated profile
+    ---------------------------
+
+    If you are truncating the profile, the optional parameter, `flux_untruncated`, specifies
+    whether the `flux` and `half_light_radius` specifications correspond to the untruncated
+    profile (`True`) or to the truncated profile (`False`, default).  The impact of this parameter
+    is a little subtle; please see the detailed description in the `Sersic` profile documentation
+    for examples.
+
+    Methods
+    -------
+
+    In addition to the usual GSObject methods, Spergel has the following access methods:
```

Comment:
Probably stupid, but this is weirding me out, what with David Spergel being a person who I know.  How about "the Spergel class"?  (To be fair, if you were to replace "Spergel" with, say, "Exponential" I wouldn't love it - I would rather say "the Exponential class" in that case too.  So I guess it's not just the name thing.)